### ISSUE TITLE
per-chunk goal setting

### ISSUE THREAD

**Issue by cloudweavers** [lizardfs/lizardfs#506]
Dear friends, we are working on adding an external tiering system for Lizard, to allow two speeds of disks (ssd and hdd to begin with, but it can be generalized to more) and automated movements between them. It would greatly help us a command to set a goal for a single chunk; do you think it would be possible?

**Comment by borkd** [comment]
@cloudweavers - Can you share more details on your approach and what do you have in the works?
Just as setting the replication goal takes care of the redundancy, having a way to selectively set affinity or "pin"  chunks to a given class of chunkservers could enable tiering out of band using tools like https://github.com/cea-hpc/robinhood

**Comment by cloudweavers** [comment]
Out of band tiering is what we are doing right now, but it's crude and presents problems (when our daemon moves a chunk from hdd to sdd and at the same time you get a chunkserver write, you end up with a conflict). Also, a per-chunk goal allows to do this form of tiering even if a server does not have SSDs (but another one does).

**Comment by psarna** [comment]
It is a little bit more complicated than it seems. Goals are right now kept inside inodes, which are associated with files, not individual chunks. We already have a big pile of functionalities waiting to be merged for 3.11.0 release and it wouldn't be wise to start yet another change before these are reviewed, tested and merged. This per-chunk-goal involves changes in core data structures of the master server (not complicated ones, but still), so we won't be introducing it right before new release.

**Comment by cloudweavers** [comment]
Of course, moving it afterwards (if it is useful for others, that is) is fine.

**Comment by borkd** [comment]
Worth the wait. Thanks!
